### PR TITLE
Adds updated text classification example with Colab notebooks

### PR DESCRIPTION
**Pull request by dantreiman**
Will update colab links to main branch after this is merged.  So, don't delete feature branch after merging.

### CODE REVIEW

File: docs/examples/text_classification.md [hunk 1]
```diff
@@ -1,31 +1,163 @@
...
-        level: word
         encoder: parallel_cnn
-
 output_features:
     -
         name: class
         type: category
+preprocessing:
+    force_split: true
+    split_probabilities: [0.7, 0.1, 0.2]
```
Comment (w4nderlust) [inline docs/examples/text_classification.md:74]: if the dataset already contains the split why are we forcing a different split?

File: docs/examples/text_classification.md [hunk 2]
```diff
@@ -1,31 +1,163 @@
...
+We'll be using AG's news topic classification dataset, a common benchmark dataset for text classification. This dataset
+is a subset of the full AG news dataset, constructed by choosing the four largest classes from the original corpus. Each
+class contains 30,000 training samples and 1,900 testing samples. The total number of training samples is 120,000 with
+7,600 total testing samples.
+
+This dataset contains four columns:
+
+| column      | description                                                                    |
+| ----------- | ------------------------------------------------------------------------------ |
+| class_index | An integer from 1 to 4: "world", "sports", "business", "sci_tech" respectively |
```
Comment (w4nderlust) [inline docs/examples/text_classification.md:17]: we have changed this in the preproc right? there's not integer anymore

Comment (dantreiman) [reply docs/examples/text_classification.md:17]: I added column `class` in preprocessing, but I didn't replace class_index.

File: docs/examples/text_classification.md (hunk 1 above)
Comment (dantreiman) [reply docs/examples/text_classification.md:74]: This dataset doesn't have a val split (only train and test), and ludwig visualize will error out if there are no validation stats.

Later on I call `ludwig evaluate` with `--split test`, and I realize it is ambiguous which split this is using.
I was hoping nobody would notice this, but since you did it should definitely be fixed :-).

Is there a way to tell Ludwig to auto-split some portion of the training set to use as the val set for training?  Or should I add a val split in the dataset preprocessing?

Comment (w4nderlust) [reply docs/examples/text_classification.md:74]: There's no way to do the split of training into training+split right now, but it's definitely a feature we need to add, let's add a task in the project page for it. I'd say in the mean time you can figure out what the best solution is ;)

Comment (dantreiman) [reply docs/examples/text_classification.md:74]: https://github.com/ludwig-ai/ludwig/pull/1868

File: docs/examples/text_classification.md [hunk 3]
```diff
@@ -1,31 +1,161 @@
...
+
+```bash
+ludwig predict \
+    --model_path results/experiment_run/model \
+    --dataset text_to_predict.csv \
+    --output_directory predictions
+```
+
+This command will write predictions to the `predictions` directory. Predictions outputs are written in multiple formats,
+including csv and parquet. For instance, `predictions/predictions.parquet` contains the predicted classes for eac
```
Comment (justinxzhao) [inline docs/examples/text_classification.md:154]:
```suggestion
including csv and parquet. For instance, `predictions/predictions.parquet` contains the predicted classes for each
```

**Review by justinxzhao — APPROVED** [review]
LGTM, looks great!

**Comment by w4nderlust** [comment]
Looked at it, and it seems great!

There are a few things i would improve, but that can be a task for the future (let's make sure we track those future improvements in the project board!):
- the model already converges in one epoch, so I would train with a smaller learning rate or report eval stats more often (every 4th of an epoch for instance), that would give better learning curve plots
- after we work on category output features to remove UNK, we should update the outputs and the confusion matrix
- after we modify the train/vali split logic we should recreate the notebook
